buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java

```diff
@@ -150,50 +199,59 @@ public PooledByteBufAllocator(boolean preferDirect, int nHeapArena, int nDirectA
 
         if (nHeapArena > 0) {
             heapArenas = newArenaArray(nHeapArena);
+            List<PoolArenaMetric> metrics = new ArrayList<PoolArenaMetric>(heapArenas.length);
             for (int i = 0; i < heapArenas.length; i ++) {
-                heapArenas[i] = new PoolArena.HeapArena(this, pageSize, maxOrder, pageShifts, chunkSize);
+                PoolArena.HeapArena arena = new PoolArena.HeapArena(this, pageSize, maxOrder, pageShifts, chunkSize);
+                heapArenas[i] = arena;
+                metrics.add(arena);
             }
+            heapArenaMetrics = Collections.unmodifiableList(metrics);
         } else {
             heapArenas = null;
+            heapArenaMetrics = Collections.emptyList();
         }
 
         if (nDirectArena > 0) {
             directArenas = newArenaArray(nDirectArena);
+            List<PoolArenaMetric> metrics = new ArrayList<PoolArenaMetric>(directArenas.length);
             for (int i = 0; i < directArenas.length; i ++) {
-                directArenas[i] = new PoolArena.DirectArena(this, pageSize, maxOrder, pageShifts, chunkSize);
+                PoolArena.DirectArena arena = new PoolArena.DirectArena(
+                        this, pageSize, maxOrder, pageShifts, chunkSize);
+                directArenas[i] = arena;
+                metrics.add(arena);
             }
+            directArenaMetrics = Collections.unmodifiableList(metrics);
         } else {
             directArenas = null;
+            directArenaMetrics = Collections.emptyList();
         }
     }
 
+    @Deprecated
+    @SuppressWarnings("UnusedParameters")
+    public PooledByteBufAllocator(boolean preferDirect, int nHeapArena, int nDirectArena, int pageSize, int maxOrder,
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the missing @deprecated Javadoc tag. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Constructor has 9 parameters, which is greater than 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS00107)